critiquebrainz/db/users.py

```diff
@@ -0,0 +1,51 @@
+from hashlib import md5
+import sqlalchemy
+from critiquebrainz import db
+
+
+def avatar(user):
+    """Args:
+        user: user data (dict or str)
```

Comment:
This is still not enough. What is user data? If you want to implement it like that, then the argument must be documented. Don't make people read the function to guess the type and structure you require.

Here's an example:
```python
"""Generates a Gravatar image URL based on user data.

Args:
    user: Dictionary with user data: {
            "id": <user ID (int)>,
            "show_gravatar": <wether this user wants their email address used to get Gravatar (bool)>
            "email": <email address (string)>,
        }

Returns:
    Generated URL of the Gravatar image for this user.
```

Said this before: https://github.com/metabrainz/critiquebrainz/pull/73#discussion_r96002699:

> Also specify exactly which data is required.

*I don't think it's a good idea to accept different types of `user` in this case. Settle on one.*